### PR TITLE
Updated voeventclient to cycle through hosts (2)

### PR DESCRIPTION
**Pull request by axj336**
I have forked @varunbhalerao original [merge request](https://github.com/lpsinger/pygcn/pull/11) and fixed some open issues that were preventing merge. 

This code ran stability for 2 days and detected LVC Alerts for candidate S200115j, S200116ah and all test events.

### CODE REVIEW

File: gcn/voeventclient.py [hunk 1]
```diff
@@ -215,11 +240,19 @@ def listen(host="68.169.57.253", port=8099,
...
     Note that this function does not return."""
+
     if log is None:
         log = logging.getLogger('gcn.listen')
 
-    while True:
-        sock = _open_socket(host, port, iamalive_timeout,
+    host, port = _validate_host_port(host, port)
+    hostcount = 0
+    num_hosts = len(host)
```
Comment (lpsinger) [inline gcn/voeventclient.py]: Please remove these two unused variables.

File: gcn/voeventclient.py [hunk 2]
```diff
@@ -215,11 +240,19 @@ def listen(host="68.169.57.253", port=8099,
...
+
+    for this_host, this_port in itertools.cycle(zip(host, port)):
+        log.info("Connecting to host: {}, port: {}".format(
```
Comment (lpsinger) [inline gcn/voeventclient.py]: Please remove, since there's already a log message that prints the host and port in `_open_socket`.

File: gcn/voeventclient.py [hunk 3]
```diff
@@ -191,7 +194,29 @@ def _ingest_packet(sock, ivorn, handler, log):
...
+def _validate_host_port(host, port):
+    """
+    Check if the host and port values are consistent with each other,
+    to be used as pairs.
+    `host` can be a string or a list of strings
+    `port` can be an integer or a list of the same length as host
+    """
+
+    if isinstance(host, six.string_types):
+        host = [host]
```
Comment (lpsinger) [inline gcn/voeventclient.py]: Please make the function able to handle the either, both, or neither of the arguments being collections.

File: gcn/voeventclient.py [hunk 4]
```diff
@@ -55,6 +57,7 @@ def _open_socket(host, port, iamalive_timeout, max_reconnect_timeout, log):
     Double the timeout after each failed attempt thereafter, until the
     timeout reaches MAX_RECONNECT_TIMEOUT. Return the new, connected socket."""
     reconnect_timeout = 1
+
```
Comment (lpsinger) [inline gcn/voeventclient.py]: Please remove unnecessary whitespace change.

File: gcn/voeventclient.py [hunk 5]
```diff
@@ -215,11 +240,19 @@ def listen(host="68.169.57.253", port=8099,
     logger will be used.
 
     Note that this function does not return."""
+
```
Comment (lpsinger) [inline gcn/voeventclient.py]: In my code style, I don't have a blank line between the docstring and the first statement in the function. Please remove the extra whitespace.

**Comment by lpsinger** [comment]
@axj336, are you able to finish up the last round of comments? Or should I finish them?

File: gcn/voeventclient.py (hunk 3 above)
Comment (axj336) [reply gcn/voeventclient.py]: I'm not sure what you mean here. Happy for you to make the change and I will rerun long term testing.

Comment (lpsinger) [reply gcn/voeventclient.py]: I'd like it to be able to handle the following cases:

* 1 host, 1 port
* 1 host, n ports
* n hosts, 1 port
* n hosts, n ports
* n hosts, m ports (ValueError)
